feat(billing): sort criteria dropdown alphabetically

sjdevgo opened the pull request:
<!--Thanks for sending a pull request! 
Please create an issue at https://github.com/openemr/openemr/issues/new/choose and then
-->

<!-- add that issue number that is fixed by this PR (In the form Fixes #123) -->
Fixes #8460 

#### Short description of what this resolves:Improves the usability of the "Choose Criteria" dropdown in the Billing Manager by sorting the list of criteria alphabetically. This helps users more easily locate filtering options when managing billing reports.


#### Changes proposed in this pull request:Added alphabetical sorting for the $TPSCriteriaDisplayMaster array in interface/billing/billing_report.php

#### Does your code include anything generated by an AI Engine? Yes 

#### If you answered yes: Verify that each file that has AI generated code has a description that describes what AI engine was used and that the file includes AI generated code.  Sections of code that are entirely or mostly generated by AI should be marked with a comment header and footer that includes the AI engine used and stating the code was AI.

sjdevgo committed:
Sort 'Choose Criteria' dropdown alphabetically in Billing Manager

stephenwaite commented on interface/billing/billing_report.php:
```diff
@@ -659,6 +669,13 @@ function criteriaSelectHasValue(select) {
+});
+$TPSCriteriaDisplayMaster = array_column($combined, 0);
+$TPSCriteriaKeyMaster = implode(",", array_column($combined, 1));
+$TPSCriteriaDataTypeMaster = implode(",", array_column($combined, 2));
```
hi @sjdevgo, thanks for the PR! It will be a valuable contribution.

Unfortunately the code below here has hard coded logic for the radio types so it's going to not associate correctly. Maybe you can figure out a way to get around this since hard coding isn't a very good solution?

kojiromike committed:
Merge master into sort-criteria-dropdown

kojiromike committed:
fix(billing): preserve radio option associations when sorting criteria

The original sorting logic sorted display names, keys, and data types
together, but left the radio option arrays hardcoded by their original
positions. This caused radio options (e.g., "All/eClaims/Paper") to be
associated with the wrong criteria after sorting.

Fix by:
- Defining radio options keyed by their database field name
- Building a single criteria list with all fields
- Sorting once and rebuilding all arrays including radio options

This also removes the code duplication between daysheet and non-daysheet
branches.

AI: Claude Opus 4.5 generated this fix based on the approach
outlined in the original PR and review feedback.
